From 23bf4877958d213066c99eeff88dc0e3a908cff9 Mon Sep 17 00:00:00 2001
From: Ron <ron@debian.org>
Date: Sun, 4 Nov 2012 14:37:27 +1030
Subject: [PATCH] Ensure the filter length is a multiple of 8

Fixes a bug where the SSE implementation would read past the end of the
buffer if it wasn't.

Also taken from f426ab5260fe9f1618a187fc2a8264dbba4aef08 in opus-tools.
---
 libspeex/resample.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libspeex/resample.c b/libspeex/resample.c
index cb3e7b4..576e782 100644
--- a/libspeex/resample.c
+++ b/libspeex/resample.c
@@ -561,8 +561,8 @@ static void update_filter(SpeexResamplerState *st)
       st->cutoff = quality_map[st->quality].downsample_bandwidth * st->den_rate / st->num_rate;
       /* FIXME: divide the numerator and denominator by a certain amount if they're too large */
       st->filt_len = st->filt_len*st->num_rate / st->den_rate;
-      /* Round down to make sure we have a multiple of 4 */
-      st->filt_len &= (~0x3);
+      /* Round up to make sure we have a multiple of 8 for SSE */
+      st->filt_len = ((st->filt_len-1)&(~0x7))+8;
       if (2*st->den_rate < st->num_rate)
          st->oversample >>= 1;
       if (4*st->den_rate < st->num_rate)
-- 
1.7.2.5

